### ISSUE TITLE
DNS resolver without sudo and systemd

### ISSUE THREAD

**Issue by benbergner** [rom1504/img2dataset#361]
I currently want to download laion400m, but quickly found out that a high performance DNS resolver like bind9 or the knot resolver is required to avoid high failure rates when downloading. However, In the HPC I'm using, there are no sudo access rights. Furthermore, running docker container conflicts with systemd, so running systemctl-commands is also not an option. I am thus not able to follow the instructions given here: https://github.com/rom1504/img2dataset#setting-up-a-high-performance-dns-resolver

Are there any best practices or alternatives for such situations, so that I can still download laion400m?

Thanks in advance

**Comment by rom1504** [comment]
Hi,
Maybe you can ask your HPC admins?

**Comment by benbergner** [comment]
Thanks, I will consider this.

I also played with `--retries`. Using the default of 0, I got a success rate of 56% (using 64 processes and 128 threads). Setting it to 1 increases success rate to 77% while still being pretty fast. Success rate saturates quite quickly at 78% for 10 retries. I wonder what the current maximum success rate is for laion400m, given that many images are down already.

Any further pointers in terms of settings (besides using a different DNS resolver) would still be much appreciated. Other than that, I think this issue can be closed.
